### DISCUSSION TITLE
Regex manager where the version number appears twice

### DISCUSSION THREAD

**Discussion by mwgamble** [renovatebot/renovate#19131] [Request Help] [answered]
I've been trying to write a regex manager for the Kubernetes AWS IAM Authenticator:
https://github.com/kubernetes-sigs/aws-iam-authenticator/releases

Right now the configuration looks like this:

```json
{
  "regexManagers": [
    {
      "description": "Maintain Kubernetes AWS IAM authenticator",
      "fileMatch": [
        "(^|/|\\.)Dockerfile$"
      ],
      "matchStrings": [
        "https://github\\.com/(?<depName>kubernetes-sigs/aws-iam-authenticator)/releases/download/(?<currentValue>[^/]+)/aws-iam-authenticator_(?<currentValue>[^_]+)_"
      ],
      "datasourceTemplate": "github-releases"
    }
  ]
}
```

This is because the version number appears twice in the download URL, and for Renovate to successfully keep it up to date, it will need to update both references.

Unfortunately, this doesn't work. That makes sense - it's not clear how any regular expression parser would handle a named capture group showing up twice.

I know that I could solve this by adding a layer of indirection, something like this:

```
# renovate: datasource=github-releases depName=kubernetes-sigs/aws-iam-authenticator
AUTHENTICATOR_VERSION=0.5.10
curl -s -o /tmp/authenticator https://github\\.com/kubernetes-sigs/aws-iam-authenticator/releases/download/v0.5.10/aws-iam-authenticator_0.5.10_linux_amd64
```

However that would require me to update all existing download commands to make use of it. I was hoping to roll something out with a Renovate configuration update to avoid requiring developers to manually update their systems.

**Comment by viceice** [comment, marked as answer]
you need to use two matchStrings regexes. renovate will be smart enough to update both in same PR

**Reply by mwgamble** [reply]
Thanks, that makes total sense :)
